backend/estimate-service/src/main/java/org/egov/works/util/LocationUtil.java

```diff
@@ -0,0 +1,174 @@
+package org.egov.works.util;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import digit.models.coremodels.RequestInfoWrapper;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.StringUtils;
+import org.egov.common.contract.request.RequestInfo;
+import org.egov.mdms.model.MasterDetail;
+import org.egov.mdms.model.MdmsCriteria;
+import org.egov.mdms.model.MdmsCriteriaReq;
+import org.egov.mdms.model.ModuleDetail;
+import org.egov.works.config.EstimateServiceConfiguration;
+import org.egov.works.repository.ServiceRequestRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import java.util.ArrayList;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+
+import static org.egov.works.util.EstimateServiceConstant.*;
+
+@Component
+@Slf4j
+public class LocationUtil {
+
+    public static final String PLACEHOLDER_CODE = "{code}";
+    public static final String filterHierarchyType = "$.[?(@.hierarchyType.code=='{code}')].boundary";
+    public static final String filterChildrenCode = "[?(@.code=='{code}')].children";
+    public static final String CODE = "code";
+    @Autowired
+    private EstimateServiceConfiguration serviceConfiguration;
+    @Autowired
+    private ServiceRequestRepository serviceRequestRepository;
+    @Autowired
+    private ObjectMapper mapper;
+
+    /**
+     * @param tenantId
+     * @param requestInfo
+     * @param hierarchyTypeCode
+     * @param boundaryLabel
+     * @param boundaryCode
+     */
+
+    public void getLocationFromLocationService(String tenantId, RequestInfo requestInfo, String hierarchyTypeCode, String boundaryLabel, String boundaryCode) {
+
+        if (StringUtils.isNotBlank(tenantId)) {
+
+            StringBuilder uri = new StringBuilder(serviceConfiguration.getLocationHost());
+
+            uri.append(serviceConfiguration.getLocationContextPath())
+                    .append(serviceConfiguration.getLocationEndpoint());
+            uri.append("?").append("tenantId=").append(tenantId);
+
+            if (StringUtils.isNotBlank(hierarchyTypeCode)) {
+                uri.append("&").append("hierarchyTypeCode=").append(hierarchyTypeCode);
+            }
+
+            if (StringUtils.isNotBlank(boundaryLabel)
+                    && StringUtils.isNotBlank(boundaryCode)) {
+                uri.append("&").append("boundaryType=").append(boundaryLabel)
+                        .append("&").append("codes=").append(boundaryCode);
+            }
+
+            Object locationRes = serviceRequestRepository.fetchResult(uri, RequestInfoWrapper.builder().requestInfo(requestInfo).build());
+
+            //TODO
+            if (locationRes != null) {
+
+            }
+            /*if (locationRes!=null) {
```

Comment:
Remove unused code. 
If this code is important, mention a comment above it.

Reply:
done.